backend/src/main/java/eu/clarin/switchboard/core/LinkMetadata.java

```diff
@@ -180,17 +182,13 @@ private static void tryToSetFilenameFromUrl(LinkInfo linkInfo, URL url) {
             path = path.substring(0, path.length() - 1);
         }
 
-        String mediatype = URLConnection.guessContentTypeFromName(path);
-        if (mediatype == null) {
-            return;
-        }
-
         String filename = path;
         int lastSlash = path.lastIndexOf("/");
         if (lastSlash >= 0) {
             filename = path.substring(lastSlash + 1, path.length());
         }
-        if (!filename.isEmpty()) {
+
+        if (filename.contains(".")) {
```

Comment:
In the end, what happens for filenames without a "." ? e.g. our [license file](https://raw.githubusercontent.com/clarin-eric/switchboard/master/LICENCE)

Reply:
I guess it remains null... which at one point was an issue but that should be resolved now (see #131). In terms of clarity of the code, I think it would be preferable to make such methods more 'functional', i.e. do not mutate incoming objects but return a value and set it to the target property where the method is called.

Reply:
I tried https://raw.githubusercontent.com/clarin-eric/switchboard/master/LICENCE against beta-switchboard. Not sure why, but it does not work. It also looks like in case of a wrong format detection && no tools for that wrong format available, the switchboard does not proceed to the tools list screen, thus making it impossible to select the right format

Reply:
I reorganized the extraction of filenames, hopefully now it makes more sense. @andmor- the LICENCE mediatype is a profiler error: https://github.com/weblicht/profiler/issues/5.

Reply:
In general, case of a profiler error shouldn't the UI also allow for manual override of the profiler input?